pkg/agent/config/config.go

```diff
@@ -461,8 +461,7 @@ func get(ctx context.Context, envInfo *cmds.Agent, proxy proxy.Proxy) (*config.N
 	if !envInfo.Debug {
 		nodeConfig.Containerd.Log = filepath.Join(envInfo.DataDir, "agent", "containerd", "containerd.log")
 	}
-	nodeConfig.Containerd.State = "/run/k3s/containerd"
-	nodeConfig.Containerd.Address = filepath.Join(nodeConfig.Containerd.State, "containerd.sock")
+	getContainerdStateAndAddress(nodeConfig)
```

Comment:
Are we getting it? It looks like we're setting it on the passed in nodeConfig, if you call the method getX I expect X to be returned.